doc/main/api/celeritas.rst

```diff
@@ -247,6 +247,105 @@ as properties of any secondary particles produced.
 .. doxygenclass:: celeritas::SBEnergyDistribution
 .. doxygenclass:: celeritas::detail::SBPositronXsCorrector
 
+.. _api_optical_physics:
+
+Optical physics
+---------------
+
+As with EM physics, the optical physics models in Celeritas are closely related
+to those in Geant4. Unlike Geant4, optical photon generation and stepping in
+Celeritas takes place in a drastically different manner.
+
+Here is a flowchart depicting the creation of optical photons in the optical
+tracking loop:
+
+.. mermaid::
+
+   flowchart TB
+     gun["Gun or external"]
+     geant4-direct["Direct Geant4 offload"]
+     geant4-scint["Geant4 scintillation"]
+     geant4-ceren["Geant4 cerenkov"]
+
+     classDef not-impl stroke-width:2px,stroke-dasharray: 5 5
+     class geant4-direct,geant4-scint,geant4-ceren not-impl
+
+     subgraph main-celeritas-loop["Main celeritas loop"]
+       offload-gather
+       scintillation-offload
+       cerenkov-offload
+     end
+
+     offload-gather -->|pre-step| scintillation-offload
+     offload-gather -->|pre-step| cerenkov-offload
+
+     subgraph photon-gen["Optical photon generation"]
+       scintillation-gen
+       cerenkov-gen
+     end
+
+     scintillation-offload -->|generator dist| scintillation-gen
+     cerenkov-offload -->|generator dist| cerenkov-gen
+     geant4-scint -->|generator dist| scintillation-gen
+     geant4-ceren -->|generator dist| cerenkov-gen
+
+
+     photons["Optical tracking loop"]
+     gun -->|primaries| photons
+
+     geant4-direct -->|primaries| photons
+     scintillation-gen -->|primaries| photons
+     cerenkov-gen -->|primaries| photons
+
+
+Optical materials
+~~~~~~~~~~~~~~~~~
+
+Optical volume and surface properties are imported from Geant4 (see below).
```

Comment:
Should we discuss our plan on having a better generalization overall? We have generalized the number of scintillation components for both materials and particle types, but arbitrary limitations that we may find on other optical processes will be redesigned as well if that provides more capabilities to the users.

Reply:
There's plenty of stuff we *could* put in, but my priority for this PR is documenting what we *have*, then next the stuff we *will*, and eventually what we *could* 😅